compute/cloud-client/src/main/java/compute/ListImages.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package compute;
+
+// [START compute_images_list]
+// [START compute_images_list_page]
+
+import com.google.cloud.compute.v1.Image;
+import com.google.cloud.compute.v1.ImagesClient;
+import com.google.cloud.compute.v1.ImagesClient.ListPage;
+import com.google.cloud.compute.v1.ListImagesRequest;
+import java.io.IOException;
+// [END compute_images_list_page]
+// [END compute_images_list]
+
+public class ListImages {
+
+  public static void main(String[] args) throws IOException {
+    // TODO(developer): Replace these variables before running the sample
+    // project: project ID or project number of the Cloud project you want to list images from.
+    String project = "your-project-id";
+    listImages(project);
+
+    // page_size: size of the pages you want the API to return on each call.
+    int pageSize = 2;
+    listImagesByPage(project, pageSize);
+  }
+
+  // [START compute_images_list]
+  // Prints a list of all non-deprecated image names available in given project.
+  public static void listImages(String project) throws IOException {
+    // Initialize client that will be used to send requests. This client only needs to be created
+    // once, and can be reused for multiple requests. After completing all of your requests, call
+    // the `instancesClient.close()` method on the client to
+    // safely clean up any remaining background resources.
+    try (ImagesClient imagesClient = ImagesClient.create()) {
+
+      // Listing only non-deprecated images to reduce the size of the reply.
+      ListImagesRequest imagesRequest = ListImagesRequest.newBuilder()
+          .setProject(project)
+          .setMaxResults(2)
```

Comment:
do we need to set this? If this example is showing the default, should we be setting parameters like this?

Reply:
The idea of this snippet is to demo the abstraction involved in fetching images even though the maxResults is set. Even though, the maxResult here is set to 2 and 2 images are fetched per call, this is abstracted when the images are listed.

Reply:
How many images are fetched per call without setting this? Would it be valuable to show both? I am just a little confused having one example explicitly setting `maxResults`, and one arbitrarily picking 2.

Reply:
If the method is not set, there would be a single abstracted call, which will fetch the images.
By setting setMaxResults, we are explicitly mentioning the number of images to be fetched for each page and inorder to get additional images, we can use the next page token.

But here, that complexity is abstracted in the code sample, even though the underlying mechanism will involve calling multiple next page tokens. 

The first method abstracts in entirety the pagination concept, whereas the second method shows how to iterate with the returned pages.